### PR TITLE
arch/risc-v/litex: Allow FDT to be passed from previous boot change.

### PR DESCRIPTION
**Pull request by g2gps**
## Summary

Allows a flattened device tree to be passed from either openSBI or the LiteX bios. The FDT is registered can be used, if supported.

## Impact

The Litex framework has tooling to generate device trees for the chosen configuration.  NuttX can use the compiled FDT to dynamically register hardware peripherals. 

This begins to address issues mentioned here:

 - https://github.com/apache/nuttx/issues/1020

And provide a better solution to the workarounds in:
 - https://github.com/apache/nuttx/pull/8233
 - https://github.com/apache/nuttx/pull/8225

As well as this being mentioned in:
 - https://github.com/apache/nuttx/pull/2844
 - https://github.com/apache/nuttx/pull/638

The long term-plan is to roll out the device tree integration used internally, to the LiteX drivers, where needed. This greatly improves the NuttX support for LiteX, which is essentially broken unless configured in exactly one way. See the limitation section on the LiteX CI repository https://github.com/enjoy-digital/litex_hw_ci/tree/main/nuttx

## Testing

Internal testing on our hardware.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Nice, let's how to make the fdt usage inside litex/qemu general enough for other chip/board.

**Review by anchao — APPROVED** [review]

**Comment by AndrewD** [comment]
@enjoy-digital this is the start of the litex device tree integration we discussed.

File: arch/risc-v/src/litex/Make.defs [hunk 1]
```diff
@@ -62,3 +62,8 @@ endif
 ifeq ($(CONFIG_LITEX_PWM),y)
 CHIP_CSRCS += litex_pwm.c
 endif
+
+ifeq ($(CONFIG_LIBC_FDT),y)
+CFLAGS += ${INCDIR_PREFIX}$(TOPDIR)$(DELIM)libs$(DELIM)libc$(DELIM)fdt$(DELIM)dtc$(DELIM)libfdt
```
[the remark this answers is not in the document]
Comment (g2gps) [reply arch/risc-v/src/litex/Make.defs]: You're right, this was a side-effect of our out-of-tree board configuration.

**Comment by g2gps** [comment]
I've added some brief documentation on how to build the FDT for NuttX as well.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**